lib/response-builder.ts

```diff
@@ -28,18 +28,41 @@ function buildResponse(
   const applicationScans: types.ScannedProjectCustom[] =
     depsAnalysis.applicationScans || [];
 
+  const scannedProjects = [
+    {
+      packageManager: plugin.packageManager,
+      depTree: pkg,
+    },
+    ...applicationScans,
+  ];
+
+  const groupedScannedProjects = groupScannedProjects(
+    pkg.name,
+    scannedProjects,
+  );
+
   return {
     plugin,
-    scannedProjects: [
-      {
-        packageManager: plugin.packageManager,
-        depTree: pkg,
-      },
-      ...applicationScans,
-    ],
+    scannedProjects: groupedScannedProjects,
   };
 }
 
+/**
+ * Group the projects for the App+OS feature.
+ * By sharing the same fields in the meta object, projects can be treated as related.
+ */
+function groupScannedProjects(
+  imageName: string,
+  scannedProjects: types.ScannedProjectCustom[],
+): types.ScannedProjectCustom[] {
+  return scannedProjects.map((project) => {
+    project.meta = {
```

Comment:
I think it's better to check if meta is already defined, just in case we dont override anything